simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+
+        Returns:
+            Optional[List[OrbitalsState]]: The resolved references to the `OrbitalsState` sections.
+        """
+        model_system = model_systems[model_index]
+
+        # If `ModelSystem` is not representative, the normalization will not run
+        if not model_system.is_representative:
+            logger.warning(
+                f"`ModelSystem`[{model_index}] section was not found to be representative."
+            )
+            return None
+
+        # If `AtomicCell` is not found, the normalization will not run
+        atomic_cell = model_system.atomic_cell[0]
+        if atomic_cell is None:
+            logger.warning("`AtomicCell` section was not found.")
+            return None
+
+        # If there is no child `ModelSystem`, the normalization will not run
+        atoms_state = atomic_cell.atoms_state
+        model_system_child = model_system.model_system
+        if model_system_child is None:
+            logger.warning("No child `ModelSystem` section was found.")
+            return None
+
+        # We flatten the `OrbitalsState` sections from the `ModelSystem` section
+        orbitals_ref = []
+        for active_atom in model_system_child:
+            # If the child is not an "active_atom", the normalization will not run
+            if active_atom.type != "active_atom":
+                continue
+            indices = active_atom.atom_indices
+            for index in indices:
+                active_atoms_state = atoms_state[index]
+                orbitals_state = active_atoms_state.orbitals_state
+                for orbital in orbitals_state:
+                    orbitals_ref.append(orbital)
+        return orbitals_ref
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set `name` to "TB"
+        self.name = "TB"
+
+        # Resolve `type` to be defined by the lower level class (Wannier, DFTB, xTB or SlaterKoster) if it is not already defined
+        self.type = (
+            self.resolve_type(logger)
+            if (self.type is None or self.type == "unavailable")
+            else self.type
+        )
+
+        # Resolve `orbitals_ref` from the info in the child `ModelSystem` section and the `OrbitalsState` sections
+        model_systems = self.m_xpath("m_parent.model_system", dict=False)
+        if model_systems is None:
+            logger.warning(
+                "Could not find the `ModelSystem` sections. References to `OrbitalsState` will not be resolved."
+            )
+            return
+        # This normalization only considers the last `ModelSystem` (default `model_index` argument set to -1)
+        orbitals_ref = self.resolve_orbital_references(model_systems, logger)
+        if orbitals_ref is not None and self.orbitals_ref is None:
+            self.n_orbitals = len(orbitals_ref)
+            self.orbitals_ref = orbitals_ref
+
+
+class Wannier(TB):
```

Comment:
I also know of Wannier being used outside of TB, e.g. band structures: https://www.vasp.at/wiki/index.php/Si_bandstructure

Reply:
That is also the same as here. VASP2WANNIER90 actually calculates first the Wannier tight binding model, and then diagonalizes it along the same path as the DFT bands.

What I mean is that there is no sense of "Wannier being used outside of TB, e.g. band structures". The bands are calculated from a Wannier TB model.

Reply:
Interesting! Thank you for clarifying that.